_ap/17/index.adoc

```diff
@@ -0,0 +1,60 @@
+---
+num: 17
+tags:
+  - "tls"
+  - "certificates"
+  - "dns"
+  - "Letsencrypt"
+title: "Automatic management of certificates per managed instance using Letsencrypt"
+status: "Accepted"
+authors:
+  - "Manyanda Chitimbo"
+---
+
+## Intent
+
+Define a pattern for the automatic management of certificates per managed instance using Letsencrypt
+
+## Motivation
+
+When a managed service is deployed on the data plane cluster its routes have be reachable via TLS.
+It means that a managed service deployed in the data plane has to provide a way to the clients (end users) to verify and validate the TLS connection.
+
+## Applicability
+
+This pattern applies to any service deployed on a dataplane that wants a managed service routes to be reachable via TLS
+
+## Description
+
+A managed service will use https://Letsencrypt.org[Letsencrypt] to managed certificates on the fly i.e automatic and thus removing the need for manual intervention and the toil related to this.
+
+To avoid sharing of certificates between deployed instances, managed service implementors are recommended to generate a wildcard certificate per managed instance.
```

Comment:
is it not wildcard or dedicated certificate?

Reply:
The AP was written in a way that there will be a dedicated wildcard certificate for all routes of a managed instance.
For example,  if a managed instance has 5 routes, all those routes will be using one dedicated wildcard certificate specific to that instance. This is the case of Kafkas for example.

Another possibility and even more fine grained approach is to have a dedicated certificate per route. Which is what I think you are suggesting to include in https://github.com/bf2fc6cc711aee1a0c2a/architecture/pull/84#discussion_r1109437233 - I'll update the AP to mention this as well.